src/seedu/addressbook/storage/AddressBookDecoder.java

```diff
@@ -0,0 +1,110 @@
+package seedu.addressbook.storage;
+
+import static seedu.addressbook.parser.Parser.PERSON_DATA_ARGS_FORMAT;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.storage.StorageFile.StorageOperationException;
+
+/**
+ * Decodes the storage data file into an AddressBook object.
+ */
+public class AddressBookDecoder {
+
+    /**
+     * Decodes the AddressBook in the storage data file.
+     *
+     * @param encodedAddressBook strings to be decoded.
+     * @return decoded AddressBook.
+     * @throws IllegalValueException if any field is invalid.
+     * @throws StorageOperationException if cannot decode any.
+     */
+    public static AddressBook decodeAddressBook(List<String> encodedAddressBook)
+            throws IllegalValueException, StorageOperationException {
+        final List<Person> successfullyDecodedPersons = decodePersonsFromStrings(encodedAddressBook);
```

Comment:
Can we inline `decodePersonsFromStrings`? This method `decodeAddressBook` doesn't seem to be doing much currently.